test/selectBuildConfigs.test.ts

```diff
@@ -0,0 +1,63 @@
+import { getWorkspacePath } from './utils/workspaceUtils';
+
+interface ISelectBuildConfigTestCase {
+    workflowFileName: string;
+    buildConfig: BuildConfig;
+    expectedSelection: undefined | {
+        line: number;
+        startChar: number;
+        endChar: number;
+    }
+}
+
+suite('Select Build Configurations in GitHub Workflow Files', () => {
+    const testCases: ISelectBuildConfigTestCase[] = [
+        { workflowFileName: 'workflow-simple.yml', buildConfig: 'api_location', expectedSelection: { line: 30, startChar: 24, endChar: 26 } },
+        { workflowFileName: 'workflow-simple.yml', buildConfig: 'app_location', expectedSelection: { line: 29, startChar: 24, endChar: 27 } },
+        { workflowFileName: 'workflow-simple.yml', buildConfig: 'output_location', expectedSelection: { line: 31, startChar: 27, endChar: 29 } },
+        { workflowFileName: 'workflow-simple.yml', buildConfig: 'app_artifact_location', expectedSelection: undefined },
+
+        { workflowFileName: 'workflow-funky.yml', buildConfig: 'api_location', expectedSelection: { line: 30, startChar: 24, endChar: 39 } },
+        { workflowFileName: 'workflow-funky.yml', buildConfig: 'app_location', expectedSelection: { line: 29, startChar: 24, endChar: 57 } },
+        { workflowFileName: 'workflow-funky.yml', buildConfig: 'output_location', expectedSelection: { line: 31, startChar: 27, endChar: 54 } },
+        { workflowFileName: 'workflow-funky.yml', buildConfig: 'app_artifact_location', expectedSelection: undefined },
+
+        { workflowFileName: 'workflow-old.yml', buildConfig: 'api_location', expectedSelection: { line: 30, startChar: 24, endChar: 29 } },
+        { workflowFileName: 'workflow-old.yml', buildConfig: 'app_location', expectedSelection: { line: 29, startChar: 24, endChar: 29 } },
+        { workflowFileName: 'workflow-old.yml', buildConfig: 'output_location', expectedSelection: undefined },
+        { workflowFileName: 'workflow-old.yml', buildConfig: 'app_artifact_location', expectedSelection: { line: 31, startChar: 33, endChar: 38 } },
+
+        { workflowFileName: 'workflow-duplicates.yml', buildConfig: 'api_location', expectedSelection: undefined },
+        { workflowFileName: 'workflow-duplicates.yml', buildConfig: 'app_location', expectedSelection: undefined },
+        { workflowFileName: 'workflow-duplicates.yml', buildConfig: 'output_location', expectedSelection: undefined },
+        { workflowFileName: 'workflow-duplicates.yml', buildConfig: 'app_artifact_location', expectedSelection: undefined },
+    ];
+
+    for (const testCase of testCases) {
+        const title: string = `${testCase.workflowFileName}: ${testCase.buildConfig}`;
+        const workspacePath: string = getWorkspacePath('testWorkspace');
+
+        test(title, async () => {
+            const configDocument: TextDocument = await workspace.openTextDocument(join(workspacePath, 'testWorkflows', testCase.workflowFileName));
```

Comment:
Rather than use separate yml files in the "testWorkflows" directory, I'd prefer you just put the test data in this file (like, as a `const`), for a few reasons:
1. Using the file system and/or VS Code APIs can slow tests down. It's worth it for end-to-end tests, but overkill for unit tests IMO
2. You can easily manipulate the test data with template literals and whatnot
3. Idk just seems easier to read when the test data is in the same file.

I feel like the only downside is this file can get pretty long, but it just doesn't bother me 🤷‍♂️. One way to help with that is to scope your test data down to smaller cases. For example, you might have one test running against an "actual" yml file seen in the wild, but the other tests are usually just testing specific edge cases without the full "yml" file necessary.